### DISCUSSION TITLE
Choosing where images taken in Hubs are saved

### DISCUSSION THREAD

**Discussion by ybinstock** [mozilla/hubs#3227] [General Questions & Discussion] [answered]
Is there a way to choose the location where images taken in hubs are saved? I know if I do my own hosting AWS, then there are stored in a bucket there, but what I'd like to do is have the images stored on the user's hard drive (whether a VR headset, desktop, or mobile). 

I'm looking at the camera-tool.js file, but I don't see anything relevant there.

Does anyone have any pointers?

**Comment by emclaren** [comment, marked as answer]
Hi there! This is a really good question. 

Unfortunately, this isn't something that would be easily done with Hubs Cloud out of the box. We don't have the capacity to provide support for this type of advanced customization, however, we will leave this discussion thread open for a while to see if anyone else has any feedback or suggestions. Please let us know if you are able to make any progress on this!

One existing Github discussion that describes how Hubs Cloud images are saved can be found here: https://github.com/mozilla/hubs/discussions/2950 - I hope this provides a bit of additional information that would be helpful in your explorations!

I also wanted to add a little disclaimer for any folks exploring this type of advanced customization that this could make your Hubs Cloud instance a bit more difficult to keep update in the future.

**Reply by emclaren** [reply]
I also wanted to mention (this may or may not be helpful), that we are doing an interface update that will include chat scrollback in the fairly near future (head to the channel called "Design" in Discord if you are interested in any additional information). 

Currently, when a user takes a photo there is a little chat that appears for a few seconds that you can click to open up the image in another tab.  With the new chat scrollback, the link to the photo will stay in the chat scrollbar until you leave the page which could make it easier for people to download / access their photos at the end of an event. This doesn't really address what you are asking for, but maybe it's helpful to know.

**Reply by robinkwilson** [reply]
Currently, there's no way to choose where images are saved in hubs because a direct download option isn't available. 

To download an image currently: 
You have the option to open link in a new tab, then download with a right click 
OR tweet the image 
OR if you have the discord bot bridging chat, the image is saved in chat history via discord.

**Reply by robinkwilson** [reply]
Marking emclaren's answer as answered, if you need more assistance or have more info please click "unmark as answer" button.

**Reply by ybinstock** [reply]
Thanks Robin and Emclaren for your answer.
